File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;
    
    people.sort((o1, o2) => (o1 - o2));
    
    let left = 0;
    let right = people.length - 1;
    while(left < right) {
        if(people[left] + people[right] > limit) {
            right--;
        } else {
            left++;
            right--;
        }
        answer++;
    }
    
    if(left === right) answer++;
    
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        Arrays.sort(people);
        int left = 0;
        int right = people.length - 1;
        while(left <= right) {
            int weight1 = people[left];
            int weight2 = people[right];
            
            if(weight1 + weight2 <= limit) {
                answer++;
                left++;
                right--;
            } else {
                answer++;
                right--;
            }
        }
        
        return answer;
    }
}